### PR TITLE
expression: implement vectorized evaluation for `builtinSHA1Sig`

### PR DESCRIPTION
**Pull request by mmyj**
### What problem does this PR solve? 
implement vectorized evaluation for builtinSHA1Sig, for #12106
### What is changed and how it works?
```
go test -v -benchmem -bench=BenchmarkVectorizedBuiltinEncryptionFunc -run=BenchmarkVectorizedBuiltinEncryptionFunc -args 'builtinSHA1Sig'
goos: linux
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkVectorizedBuiltinEncryptionFunc/builtinSHA1Sig-VecBuiltinFunc-4                    3430            331624 ns/op          107439 B/op       3229 allocs/op
BenchmarkVectorizedBuiltinEncryptionFunc/builtinSHA1Sig-NonVecBuiltinFunc-4                 3067            392840 ns/op          197741 B/op       4035 allocs/op
BenchmarkVectorizedBuiltinEncryptionFunc/builtinSHA1Sig-VecBuiltinFunc#01-4                 3531            333504 ns/op          107424 B/op       3233 allocs/op
BenchmarkVectorizedBuiltinEncryptionFunc/builtinSHA1Sig-NonVecBuiltinFunc#01-4              3100            386005 ns/op          197836 B/op       4040 allocs/op
PASS
ok      github.com/pingcap/tidb/expression      4.898s
```
### Check List 
Tests 
- Unit test

### CODE REVIEW

**Comment by mmyj** [comment]
what should I do to fix this error？

File: expression/builtin_encryption_vec.go [hunk 1]
```diff
@@ -171,11 +174,35 @@ func (b *builtinPasswordSig) vecEvalString(input *chunk.Chunk, result *chunk.Col
...
+	}
+	result.ReserveString(n)
+	hasher := sha1.New()
+	for i := 0; i < n; i++ {
+		if buf.IsNull(i) {
+			result.AppendNull()
+			continue
+		}
+		str := buf.GetString(i)
+		_, err = hasher.Write([]byte(str))
```
Comment (XuHuaiyu) [inline expression/builtin_encryption_vec.go]: s/ []byte(str)/ hack.Slice(str)

Comment (mmyj) [reply expression/builtin_encryption_vec.go]: I will fix it.

**Comment by XuHuaiyu** [comment]
Is the benchmark result changed? If so, it will be better to update the result in the description.

File: expression/builtin_encryption_vec.go [hunk 2]
```diff
@@ -171,11 +175,35 @@ func (b *builtinPasswordSig) vecEvalString(input *chunk.Chunk, result *chunk.Col
...
+	}
+	result.ReserveString(n)
+	hasher := sha1.New()
+	for i := 0; i < n; i++ {
+		if buf.IsNull(i) {
+			result.AppendNull()
+			continue
+		}
+		str := buf.GetString(i)
+		_, err = hasher.Write(hack.Slice(str))
```
Comment (qw4990) [inline expression/builtin_encryption_vec.go]: We can use `Column.GetBytes` here.

Comment (mmyj) [reply expression/builtin_encryption_vec.go]: ok, I will fix it now, and then update the performance compare for the new implement.

**Comment by mmyj** [comment]
the new benchmark result
```
go test -v -benchmem -bench=BenchmarkVectorizedBuiltinEncryptionFunc -run=BenchmarkVectorizedBuiltinEncryptionFunc -args 'builtinSHA1Sig'
goos: linux
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkVectorizedBuiltinEncryptionFunc/builtinSHA1Sig-VecBuiltinFunc-4         	    2868	    390919 ns/op	   90074 B/op	    2410 allocs/op
BenchmarkVectorizedBuiltinEncryptionFunc/builtinSHA1Sig-NonVecBuiltinFunc-4      	    2024	    562301 ns/op	  196654 B/op	    4015 allocs/op
BenchmarkVectorizedBuiltinEncryptionFunc/builtinSHA1Sig-VecBuiltinFunc#01-4      	    2906	    403684 ns/op	   92876 B/op	    2485 allocs/op
BenchmarkVectorizedBuiltinEncryptionFunc/builtinSHA1Sig-NonVecBuiltinFunc#01-4   	    2163	    563364 ns/op	  202588 B/op	    4140 allocs/op
PASS
ok  	github.com/pingcap/tidb/expression	4.890s

```

**Pull request merged**